src/Client.js

```diff
@@ -227,6 +244,7 @@ Client.prototype._performRequest = function(
       'X-FaunaDB-API-Version': APIVersion,
       'X-Fauna-Driver': 'Javascript',
       'X-Last-Seen-Txn': this._lastSeen,
+      'X-Query-Timeout': queryTimeout,
```

Comment:
If no `queryTimeout` is specified, would we want this header to emit?

Reply:
It is ok to have. No problems are caused passing this header as empty.

Reply:
@danieltodonnell Now, you can pass the queryTimeout value when instantiating the Client. If no value is passed, it defaults to null. Then, the `removeNullAndUndefinedValues` call on line 241 removes any headers that are `undefined` or `null`.

On line 234, we check the `options` object passed to `.query()` to see if a `queryTimeout` has been passed. If so, we use that value instead of the Client's `this._queryTimeout` 👍

Reply:
Great clarification regarding `removeNullAndUndefinedValues`. Probably warrants a comment at this point in the code too.

Reply:
Thanks for all the due diligence everyone!